### ISSUE TITLE
Failure when running magmasat calculations in a loop

### ISSUE THREAD

**Issue by simonwmatthews** [kaylai/VESIcal#151]
I tried to do some saturation pressure calculations for a series of melt inclusions by using magmasat in a for loop. After a few (<6) calculations magmastat would start taking an extremely long time, and would fail to find a saturation pressure. Other magmasat calculations (e.g. calculate_dissolved_volatiles) also wouldn't work correctly. Re-running the loop would result in none of the calculations succeeding. Restarting the kernel meant magmastat would start working again, but only for a limited number of calculations.

I noticed the same issue when doing calculations with a BatchFile. A single BatchFile run would succeed, but then if I tried to do another without restarting the kernel, the calculations would fail.

All this was done on my local machine using the latest version of thermoengine. I recreated the same result using the ENKI production server.

I realise this issue is a bit vague, and I will put together a minimally working example soon, but thought I would check to see whether anyone else has noticed this issue?

**Comment by PennyWieser** [comment]
I've had this a few times throughout. I think @kaylai tried to fix it but thermoengine just gets really unhappy?
Is it always failing at the same inclusion # after restarting the kernel?
Although its far from a perfect solution, now there is the status bar, on the enki server I find that you can see the same name, and just delete it from the input file. I normally then try running the weird deleted one on its own, very occasionally it works, mostly it doenst and mostly that composition also doesnt work on the MagmaSat App. This is something folks have complaindd about for some time with the app - in cardiff they adjust the temp a tiny bit until they get an answer.

**Comment by simonwmatthews** [comment]
Yes, I've also run into that sort of behaviour before (and it usually crashes the kernel for me), though this "new" issue seems to be subtly different. The puzzling thing is that exactly the same calculation done through a BatchFile works perfectly, and then will immediately stop working for the next BatchFile. It made me wonder whether there's some initialisation step that's happening routinely in the BatchFile methods that isn't happening in the single calculation methods....
